autoprecompiles/src/optimizer.rs

```diff
@@ -75,12 +75,16 @@ fn optimization_loop_iteration<T: FieldElement>(
     )?;
     let constraint_system = constraint_system.system().clone();
     let constraint_system = if let Some(memory_bus_id) = bus_map.get_bus_id(&BusType::Memory) {
-        let constraint_system =
-            optimize_memory(constraint_system, memory_bus_id, NoRangeConstraints);
-        assert!(check_register_operation_consistency(
-            &constraint_system,
-            memory_bus_id
-        ));
+        let constraint_system = optimize_memory::<
+            A::PowdrField,
+            AlgebraicReference,
+            A::MemoryBusInteraction,
```

Comment:
Does this work? `T` and `V` should be constrained already by the third parameter.
```suggestion
            _,
            _,
            A::MemoryBusInteraction,
```